### PR TITLE
Update tests.

### PR DESCRIPTION
**Pull request by sergeisakov**
Updates one simulator test to cover all ApplyGate* functions. Adds a multi-qubit fuser test for small circuits.

### CODE REVIEW

File: tests/fuser_mqubit_test.cc [hunk 1]
```diff
@@ -371,6 +373,269 @@ TEST(FuserMultiQubitTest, Simulation) {
...
+      CreateDummyGate(2, {5, 6}),
+      CreateDummyGate(3, {3, 4}),
+    };
+
+    {
+      param.max_fused_size = 3;
+      auto fused_gates = Fuser::FuseGates(
+          param, num_qubits, circuit.begin(), circuit.end());
+
+      EXPECT_TRUE(TestFusedGates(num_qubits, circuit, fused_gates));
```
Comment (95-martin-orion) [inline tests/fuser_mqubit_test.cc:626]: No expectation on `fused_gates.size()` here?

File: tests/fuser_mqubit_test.cc [hunk 2]
```diff
@@ -371,6 +373,269 @@ TEST(FuserMultiQubitTest, Simulation) {
...
+    }
+
+    {
+      param.max_fused_size = 5;
+      auto fused_gates = Fuser::FuseGates(
+          param, num_qubits, circuit.begin(), circuit.end());
+
+      EXPECT_TRUE(TestFusedGates(num_qubits, circuit, fused_gates));
```
Comment (95-martin-orion) [inline tests/fuser_mqubit_test.cc:634]: No expectation on `fused_gates.size()` here?

**Review by 95-martin-orion — APPROVED** [review]
The broad range of `fuser_mqubit` tests looks great! A couple of comments on those (not sure if anything needs to be changed) but otherwise this can be merged.

File: tests/fuser_mqubit_test.cc (hunk 1 above)
Comment (sergeisakov) [reply tests/fuser_mqubit_test.cc:626]: Yes, the reason here and below is that the current version of the fuser does not give the optimal number of fused gates. Perhaps some future version will be better.

**Pull request merged**